scripts/inspect.sh

```diff
@@ -184,7 +184,7 @@ function suggest_fixes {
 
   # check for docker
   # if docker is installed
-  if [ -d "/etc/docker/" ]; then 
+  if [ -d "/etc/docker/" && ! -z "$(which dockerd)" ]; then
```

Comment:
This is giving me `/snap/microk8s/x6/inspect.sh: line 187: [: missing ]`

Maybe we need to split the tests in two, see for example:
```
>  if [ -d "/etc/docker/" ] && ! [ -z "$(which ls)" ]; then echo "docker is here"; fi 
docker is here
>  if [ -d "/etc/docker/" && ! -z "$(which ls)" ]; then echo "docker is here"; fi 
bash: [: missing `]'
```